### ISSUE TITLE
Builtin analytics

### ISSUE THREAD

**Issue by sils** [Cretezy/pymessenger2#8]
I wonder if it'd make sense to give the user the ability to just give an API key for https://www.dashbot.io/ to get analytics. Possibly realize it as some kind of plugin to not endorse a commercial product, not sure, but I'm certainly convinced it's a value add and pymessenger2 would be a nice place to add it without having redundancy at the client side.

**Comment by Cretezy** [comment]
Yeah that'd be great, however we should provide a modular way -- and make it easy and open to add other analytics (Google Analytics comes up to mind).

**Comment by sils** [comment]
Yep we can just add this, if there's an API key it'll send all outgoing messages there and we can add hooks for other analytics at any point when someone needs them, I have no idea how GA would work for a bot but anything similar to dashbot will be very simple to add as well
